### PR TITLE
[MIRROR] [READY]Randomized Station Time + Night Shifts

### PR DESCRIPTION
**Pull request by CitadelStationBot**
Original PR: https://github.com/tgstation/tgstation/pull/35775
--------------------
:cl: Kevinz000 & Deathride58
experimental: Station time is now randomized [if enabled in game_options configuration]. Station time is 1:1 with realtime, compensating for server lag ofcourse because it's based on world.time but with a randomized offset added to it. Adminhelp tickets will be adjusted to compensate. Station time now also ticks faster, by a default of 24:1 so 1 hour real-time is 24 hours station-time. Subject to debate.
rscadd: A separate round time has been added to status panel. This will start at 00:00:00.
rscadd: Night shift lighting [if enabled in the same configuration] will activate between station time 7:30 PM and 7:30 AM. This will dim all lights affected, but they will still have the same range.
rscadd: APCs now have an option to set night lighting mode on or off, regardless of time.
/:cl:
Credits to @ deathride58 for the lighting range/coloration and idea.
Screenshots by @ deathride58 
Normal/daytime lighting
image

Night shift lighting
image

### CODE REVIEW

**Review by deathride58 — DISMISSED** [review]
Needs nightshift code removed from modular_citadel

File: code/controllers/subsystem/ticker.dm [hunk 1]
```diff
@@ -43,7 +43,8 @@ SUBSYSTEM_DEF(ticker)
 	var/timeLeft						//pregame timer
 	var/start_at
 
-	var/gametime_offset = 432000 // equal to 12 hours, making gametime at roundstart 12:00:00
+	var/gametime_offset = 432000		//Deciseconds to add to world.time for station time.
+	var/station_time_rate_multiplier = 24		//factor of station time progressal vs real time.
```
Comment (deathride58) [inline code/controllers/subsystem/ticker.dm:47]: Remind me to override this in modular_citadel to be 1:1 with realworld time to preserve the current nightshift behavior

**Review by Poojawa — APPROVED** [review]

File: code/controllers/subsystem/ticker.dm (hunk 1 above)
Comment (Poojawa) [reply code/controllers/subsystem/ticker.dm:47]: hey nerd, override this in modular_citadel. 

c:

Comment (Poojawa) [reply code/controllers/subsystem/ticker.dm:47]: also tbh, I like it randomized, give the daywalkers a chance to enjoy moody light

**Review by deathride58 — APPROVED** [review]
Just needs a config update.

change NIGHTSHIFT_ENABLED to ENABLE_NIGHT_SHIFTS

Remove NIGHTSHIFT_START and NIGHTSHIFT_END

And add either RANDOMIZE_SHIFT_TIME or SHIFT_TIME_REALTIME

**Comment by Poojawa** [comment]
@LetterJay 

For config things

**Pull request merged**